src/libs/NextStepUtils.ts

```diff
@@ -27,5 +50,270 @@ function parseMessage(messages: Message[] | undefined) {
+                        [CONST.POLICY.AUTO_REPORTING_FREQUENCIES.MONTHLY]: formattedDate ? `on the ${formattedDate} of each month` : '',
+                        [CONST.POLICY.AUTO_REPORTING_FREQUENCIES.TRIP]: 'at the end of your trip',
+                        [CONST.POLICY.AUTO_REPORTING_FREQUENCIES.INSTANT]: '',
+                        [CONST.POLICY.AUTO_REPORTING_FREQUENCIES.MANUAL]: '',
+                    };
+
+                    if (harvestingSuffixes[autoReportingFrequency]) {
+                        harvestingSuffix = ` ${harvestingSuffixes[autoReportingFrequency]}`;
+                    }
+                }
+
+                optimisticNextStep.message.push(
+                    {
+                        text: `automatically submit${harvestingSuffix}!`,
+                        type: 'strong',
+                    },
+                    {
+                        text: ' No further action required!',
+                    },
+                );
+            }
+
+            // Prevented self submitting
+            if (isPreventSelfApprovalEnabled && isSelfApproval) {
+                optimisticNextStep.message = [
+                    {
+                        text: "Oops! Looks like you're submitting to ",
+                    },
+                    {
+                        text: 'yourself',
+                        type: 'strong',
+                    },
+                    {
+                        text: '. Approving your own reports is ',
+                    },
+                    {
+                        text: 'forbidden',
+                        type: 'strong',
+                    },
+                    {
+                        text: ' by your policy. Please submit this report to someone else or contact your admin to change the person you submit to.',
+                    },
+                ];
+            }
+
+            break;
+
+        // Generates an optimistic nextStep once a report has been submitted
+        case CONST.REPORT.STATUS_NUM.SUBMITTED: {
+            const verb = isManager ? 'review' : 'approve';
+
+            // Self review & another reviewer
+            optimisticNextStep = {
+                type,
+                title: 'Next Steps:',
+                message: [
+                    {
+                        text: 'Waiting for ',
+                    },
+                    {
+                        text: managerDisplayName,
+                        type: 'strong',
+                    },
+                    {
+                        text: ' to ',
+                    },
+                    {
+                        text: verb,
+                        type: 'strong',
+                    },
+                    {
+                        text: ' %expenses.',
+                    },
+                ],
+            };
+
+            // Another owner
+            if (!isOwner) {
+                optimisticNextStep.message = [
+                    {
```

Comment:
Doesnt this overwrite the previous message? if you are the manager, most of the time you are not the owner either so this would always overwrite the message, no?

Reply:
Yes, it does. I got you. It makes sense to flip a condtion to avoid extra overwriting.

Reply:
Done - b9ac4a3.